src/displayapp/screens/Dice.cpp

```diff
@@ -0,0 +1,231 @@
+  lv_obj_set_pos(rollBtn, 165, 5);
+  lv_obj_set_ext_click_area(rollBtn, 5, 5, 5, 5);
+  rollBtnLabel = lv_label_create(rollBtn, nullptr);
+  lv_label_set_text(rollBtnLabel, "Roll");
+
+  numDiceIncBtn = lv_btn_create(configCont, nullptr);
+  numDiceIncBtn->user_data = this;
+  lv_obj_set_event_cb(numDiceIncBtn, num_dice_inc_event_handler);
+  lv_obj_set_size(numDiceIncBtn, 70, 50);
+  lv_obj_set_pos(numDiceIncBtn, 5, 5);
+  lv_obj_set_ext_click_area(numDiceIncBtn, 5, 5, 5, 10);
+  numDiceIncBtnLabel = lv_label_create(numDiceIncBtn, nullptr);
+  lv_label_set_text(numDiceIncBtnLabel, "+");
+
+  numDiceLabelCont = lv_cont_create(configCont, nullptr);
+  lv_obj_set_size(numDiceLabelCont, 80, 40);
+  lv_obj_set_pos(numDiceLabelCont, 0, 60);
+  lv_obj_set_style_local_bg_color(numDiceLabelCont, LV_CONT_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_BLACK);
+  lv_cont_set_layout(numDiceLabelCont, LV_LAYOUT_CENTER);
+  numDiceLabel = lv_label_create(numDiceLabelCont, nullptr);
+  lv_label_set_text_fmt(numDiceLabel, "%d", 1);
+
+  numDiceDecBtn = lv_btn_create(configCont, nullptr);
+  numDiceDecBtn->user_data = this;
+  lv_obj_set_event_cb(numDiceDecBtn, num_dice_dec_event_handler);
+  lv_obj_set_size(numDiceDecBtn, 70, 50);
+  lv_obj_set_pos(numDiceDecBtn, 5, 105);
+  lv_obj_set_ext_click_area(numDiceDecBtn, 5, 5, 10, 5);
+  numDiceDecBtnLabel = lv_label_create(numDiceDecBtn, nullptr);
+  lv_label_set_text(numDiceDecBtnLabel, "-");
+
+  diceSizeIncBtn = lv_btn_create(configCont, nullptr);
+  diceSizeIncBtn->user_data = this;
+  lv_obj_set_event_cb(diceSizeIncBtn, dice_size_inc_event_handler);
+  lv_obj_set_size(diceSizeIncBtn, 70, 50);
+  lv_obj_set_pos(diceSizeIncBtn, 85, 5);
+  lv_obj_set_ext_click_area(diceSizeIncBtn, 5, 5, 5, 10);
+  diceSizeIncBtnLabel = lv_label_create(diceSizeIncBtn, nullptr);
+  lv_label_set_text(diceSizeIncBtnLabel, "+");
+
+  diceSizeLabelCont = lv_cont_create(configCont, nullptr);
+  lv_obj_set_size(diceSizeLabelCont, 80, 40);
+  lv_obj_set_pos(diceSizeLabelCont, 80, 60);
+  lv_obj_set_style_local_bg_color(diceSizeLabelCont, LV_CONT_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_BLACK);
+  lv_cont_set_layout(diceSizeLabelCont, LV_LAYOUT_CENTER);
+  diceSizeLabel = lv_label_create(diceSizeLabelCont, nullptr);
+  lv_label_set_text_fmt(diceSizeLabel, "d%d", 6);
+
+  diceSizeDecBtn = lv_btn_create(configCont, nullptr);
+  diceSizeDecBtn->user_data = this;
+  lv_obj_set_event_cb(diceSizeDecBtn, dice_size_dec_event_handler);
+  lv_obj_set_size(diceSizeDecBtn, 70, 50);
+  lv_obj_set_pos(diceSizeDecBtn, 85, 105);
+  lv_obj_set_ext_click_area(diceSizeDecBtn, 5, 5, 10, 5);
+  diceSizeDecBtnLabel = lv_label_create(diceSizeDecBtn, nullptr);
+  lv_label_set_text(diceSizeDecBtnLabel, "-");
+}
+
+void Dice::rollDice() {
+  uint8_t roll = 0;
+  uint16_t result = 0;
+
+  lv_label_set_text(resultTotalLabel, "");
+  lv_label_set_text(resultIndRollsLabel, "");
+
+  for (int i = 0; i < numDice; i++) {
+    roll = rand() % diceSizes[diceSizeIndex] + 1;
```

Comment:
Don't think that anyone will particularly care but this implementation creates a slightly biased distribution of results for any dice sizes that are not a factor of `RAND_MAX` (which is all of the current ones).

Reply:
Interesting.  Have any advice for a different implementation?  Or do you think it's good enough as-is?

Reply:
@jonvmey @tmilburn -- just implemented a fix for this in the latest changes!